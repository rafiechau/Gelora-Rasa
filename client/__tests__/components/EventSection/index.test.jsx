// Tes untuk TopEventsSection
import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import TopEventsSection from '@components/EventSection';
import { IntlProvider } from 'react-intl';

describe('TopEventsSection', () => {
  const mockEvents = [
    { title: 'Event 1', image: '/img/event1.jpg', category: 'Category 1', date: '2023-01-01' },
    { title: 'Event 2', image: '/img/event2.jpg', category: 'Category 2', date: '2023-02-02' },
  ];
  const mockOnNavigate = jest.fn();

  beforeEach(() => {
    render(
      <IntlProvider locale="id">
        <TopEventsSection events={mockEvents} onNavigate={mockOnNavigate} />
      </IntlProvider>
    );
  });

  test('Should render correctly', () => {
    expect(screen.getByTestId('top-events-section')).toBeInTheDocument();
  });

  test('Should display all events', () => {
    mockEvents.forEach((event, index) => {
      expect(screen.getByTestId(`event-card-${index}`)).toBeInTheDocument();
    });
  });

  test('Should navigate on browse all button click', () => {
    fireEvent.click(screen.getByTestId('browse-all-button'));
    expect(mockOnNavigate).toHaveBeenCalledWith('/login');
  });

  test('Should match snapshot', () => {
    const topEventsSection = screen.getByTestId('top-events-section');
    expect(topEventsSection).toMatchSnapshot();
  });
});
