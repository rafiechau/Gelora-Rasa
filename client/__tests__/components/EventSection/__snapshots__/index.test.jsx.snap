// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopEventsSection Should match snapshot 1`] = `
<div
  class="topEventsSection"
  data-testid="top-events-section"
>
  <div
    class="topEventsHeader"
  >
    <div
      class="topEventsTitle"
      data-testid="top-events-title"
    >
      app_top_events_title
    </div>
    <button
      class="browseAllButton"
      data-testid="browse-all-button"
      type="button"
    >
      app_home_before_login_browse_button
    </button>
  </div>
  <div
    class="cardsContainer"
    data-testid="cards-container"
  >
    <div
      class="eventCard"
      data-testid="event-card-0"
    >
      <img
        alt="Event 1"
        class="cardImage"
        src="/img/event1.jpg"
      />
      <div
        class="eventTitle"
      >
        Event 1
      </div>
      <div
        class="eventCategory"
      >
        Category 1
      </div>
      <div
        class="eventDate"
      >
        2023-01-01
      </div>
    </div>
    <div
      class="eventCard"
      data-testid="event-card-1"
    >
      <img
        alt="Event 2"
        class="cardImage"
        src="/img/event2.jpg"
      />
      <div
        class="eventTitle"
      >
        Event 2
      </div>
      <div
        class="eventCategory"
      >
        Category 2
      </div>
      <div
        class="eventDate"
      >
        2023-02-02
      </div>
    </div>
  </div>
</div>
`;
